### ISSUE TITLE
Chat-Ops integration with Wiki

### ISSUE THREAD

**Issue by marceloschmidt** [RocketChat/Rocket.Chat#1669]
It would be cool to have an integration with Github/Gitlab Wiki, or an internal wiki, for storing .md files related to the room and being able to edit them on-the-fly. 
Displaying items in a tree-like structure would be good too.
##

**Comment by fallc0nn** [comment]
Great :)
A multi-user editing mechanism for .md files in a room would increase a lot of our productivity, keeping workflow entirely on rocket.chat.

Example: We can use it to keep important project's info/files direct on a project channel and update this documentation in the channel. A tree-like structure would even enhance this experience.

**Comment by Sing-Li** [comment]
JavaScript source file edit would be awesome  (although not quite what the title suggests) !

Multi-user shared edit as per @fallc0nn ?  or  Private per-user?  Very different use cases.

Also, how will we 'delegate' the identify?  What if the user did not oauth'd in via GitHub or GitLab - no identity for write on that system?

**Comment by fallc0nn** [comment]
"Multi-user shared edit as per @fallc0nn ? or Private per-user? Very different use cases."

I'm thinking more simple for now. Not simultaneous editing.

Imagine this scenario:
- A technical project internal group with 5 analysts.
- Analyst #1 creates a .md file with procedures, snippet codes from conf files etc.
- One week later, Analyst #2 updated one of the conf files and wants to update the .md main file created by Analyst #1.
- Two months later, Analyst #6 joined company and wants to read and update the .md file to make more verbose. He separates the original .md file in four .md files: "conf_files.md", "architecture.md", "contact_info.md" and "project_info.md", put all in a tree-like structure under "md" directory.
- Analyst #4 creates a "img" directory with Visio-like images of the architecture.

And so on. 

tl;dr
What I'm aiming for is a limited (but useful+productive) way to keep collaborative files from projects into a project/activity room.

**Comment by Sing-Li** [comment]
So ...  

* a managed collection of .md files associated with each channel; using current upload file storage is probably fine
* a markdown renderer and editor for these files
* a file manager to access these files
* the two bullets above can be combined if we find a JS embeddable wiki-in-a-window

**Comment by fallc0nn** [comment]
Yeah, perfect.
To address this issue today, we use a few markdown messages in the channel that only the administrator can edit, and everybody pinned to quickly keep track of last edit.

I think this request can be useful not only for us but for all.

**Comment by haceru** [comment]
If I may join the conversation, this feature would be extremely useful for collaborative document editing too (for example a procedure or a legislative proposal with multiple editors). If at the end you could export the result as a PDF for example, it would be awesome !

**Comment by engelgabriel** [comment]
Relates on #2426

**Comment by engelgabriel** [comment]
Please see initial documentation at https://github.com/RocketChat/Rocket.Chat/wiki/WebHook-Scripting---GitHub

**Comment by geekgonecrazy** [comment]
@marceloschmidt now that we've moved over to the docs repo, what would that leave us needing to do?

**Comment by marceloschmidt** [comment]
The original request was for a package or tabbar plugin that would allow for editing on-the-fly. So I think the request is still pending :)

**Comment by geekgonecrazy** [comment]
Let the chatops thinking begin!  I can see this being very easily done with the in band message PR I need to finish up.

**Comment by B-Stefan** [comment]
Maybe this is interesting for you. 
To add your own fly in panel I developed a docker container that pulls the RocketChat source code and generates a new build of RocketChat with your custom meteor module. 

**This is highly experimental** 

https://github.com/B-Stefan/Sofia/tree/master/service-rocket-chat

**Comment by geekgonecrazy** [comment]
This would make for a great Rocket.Chat App. But not something we necessarily would want to add to our core code base.

If any want to discuss or propose this feature in detail please feel free to open a post on [our forum](https://forums.rocket.chat) in the feature request category.

Thanks,
